src/libsyntax/feature_gate.rs

```diff
@@ -2214,6 +2218,10 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
                 gate_feature_post!(&self, exclusive_range_pattern, pattern.span,
                                    "exclusive range pattern syntax is experimental");
             }
+            PatKind::Or(_) => {
+                gate_feature_post!(&self, or_patterns, pattern.span,
+                                   "or_patterns syntax is experimental");
```

Comment:
This feature changes the grammar and needs to be gated at parse time, so e.g. `pat` matcher in macros doesn't start accepting ` A | B` on stable.

https://github.com/rust-lang/rust/pull/60669 has an example of feature gating done at parse time (or rather on data collected at parse time).

Reply:
:+1: thanks for the example. I pretty much copied what was done for #60669